### PR TITLE
Fix: workflow end time refill

### PR DESCRIPTION
**Pull request by Somefive**
### Description of your changes

In KubeVela v1.6, when workflow finished, it will fill the EndTime to the status of the application. Previously, this will be done every time we reached the end of workflow. It will let the application continuously update the end time of workflow even if workflow does not run anymore, for example, state-keeping or garbage-collecting. This PR fixes it.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go [hunk 1]
```diff
@@ -221,9 +221,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	isUpdate := app.Status.Workflow.Message != "" && workflowInstance.Status.Message == ""
 	workflowInstance.Status.Phase = workflowState
 	app.Status.Workflow = workflow.ConvertWorkflowStatus(workflowInstance.Status, app.Status.Workflow.AppRevision)
+	logCtx.Info("Workflow return state=%s", workflowState)
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:224]: remove this debug log

File: pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go [hunk 2]
```diff
@@ -233,33 +233,28 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}
 		return r.gcResourceTrackers(logCtx, handler, common.ApplicationWorkflowSuspending, false, isUpdate)
 	case workflowv1alpha1.WorkflowStateTerminated:
-		logCtx.Info("Workflow return state=Terminated")
-		handler.UpdateApplicationRevisionStatus(logCtx, handler.latestAppRev, false, app.Status.Workflow)
-		r.doWorkflowFinish(app, workflowState)
+		if workflowInstance.Status.EndTime.IsZero() {
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:236]: we need to remove this endtime when workflow need to re-run

**Comment by wonderflow** [comment]
do we need to backport?

**Comment by Somefive** [comment]
> do we need to backport?

This one does not need.

File: pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go (hunk 1 above)
Comment (Somefive) [reply pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:224]: This is not a debug log. It is a refactor of previous logs.

File: pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go (hunk 2 above)
Comment (Somefive) [reply pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:236]: Yes, it will. The end time will be cleared when workflow restarts.

**Review by wonderflow — APPROVED** [review]

**Pull request merged**